### PR TITLE
DR dashboard metrics changes

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Ramen fix PR: https://github.com/RamenDR/ramen/pull/784
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2177695

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
issue:
RPO_cards

File: packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx [hunk 1]
```diff
@@ -131,22 +120,21 @@ export const ProtectedPVCsSection: React.FC<ProtectedPVCsSectionProps> = ({
...
+  const updateRPO = React.useCallback(() => {
+    setRPO(
+      fromNow(selectedAppSet?.placementInfo?.[0]?.lastGroupSyncTime) || 'N/A'
     );
-    const rpo = Number(item?.value[1]);
-    return !!rpo ? humanizeRPO(rpo)?.string : 'N/A';
-  }, [selectedAppSet, lastSyncTimeData]);
+  }, [selectedAppSet]);
+
+  React.useEffect(() => {
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx:133]: Earlier, I was calculating RPO from metrics, which leads to a lot of conversion and logic getting a little bit weird, Now I am simply calculating RPO from DRPC CR's lastGroupSyncTime. Also reusing "fromNow" utils function.

File: packages/mco/components/mco-dashboard/data-policy/cluster-app-card/cluster-app-card.tsx [hunk 2]
```diff
@@ -266,7 +261,6 @@ type ClusterWiseCardProps = {
 };
 
 type AppWiseCardProps = {
-  lastSyncTimeData: PrometheusResponse;
```
Comment (GowthamShanmugam) [inline packages/mco/components/mco-dashboard/data-policy/cluster-app-card/cluster-app-card.tsx:269]: Now, none of them in app-specific card is using this metrics, but still this is used under cluster wise view.

File: packages/mco/constants/dashboard.ts [hunk 3]
```diff
@@ -5,3 +5,8 @@ export const CLUSTER_CLAIM_URL_NAME =
   'consoleurl.cluster.open-cluster-management.io';
 export const applicationDetails =
   '/multicloud/applications/details/:namespace/:name';
+export const OBJECT_NAME = 'obj_name';
+export const OBJECT_NAMESPACE = 'obj_namespace';
+export const RAMEN_HUB_OPERATOR_METRICS_SERVICE =
+  'job="ramen-hub-operator-metrics-service"';
+export const DRPC_OBJECT_TYPE = 'obj_type="DRPlacementControl"';
```
Comment (GowthamShanmugam) [inline packages/mco/constants/dashboard.ts:12]: For ref: https://github.com/RamenDR/ramen/pull/784/files#diff-8e035c75bcd87ddbc41b4c0d81c4bc0d5b30e813cb9668124f97fb6ad2370ee9R28

File: packages/shared/utils/humanize.js [hunk 4]
```diff
@@ -253,43 +254,36 @@ export const humanizePercentage = (value) => {
...
-  sec: 1,
-  min: 60,
-  hour: 3600,
-  day: 86400,
-  week: 604800,
-  month: 2629800,
-  year: 31557600,
+export const humanizeMinutes = (seconds, customUnitUnit) => {
+  const val = Number(seconds);
+  const { minutes } = getDuration(val * 1000);
```
Comment (GowthamShanmugam) [inline packages/shared/utils/humanize.js:259]: reusing getDuration utils and removed my logic

**Comment by SanjalKatiyar** [comment]
/retitle DR dashboard metrics changes - WIP
now we have 4.13 branches, we need to add BZ number to the title for backport PRs now... not for master...

File: packages/mco/utils/disaster-recovery.tsx [hunk 5]
```diff
@@ -365,7 +365,7 @@ export const getSLAStatus = (
   );
   const slaDiff = slaTakenInSeconds / syncIntervalInSeconds || 0;
   if (slaDiff >= THRESHOLD) return [SLA_STATUS.CRITICAL, slaDiff];
-  else if (slaDiff > 1 && slaDiff < THRESHOLD)
+  else if (slaDiff > 2 && slaDiff < THRESHOLD)
```
Comment (SanjalKatiyar) [inline packages/mco/utils/disaster-recovery.tsx:368]:
```suggestion
  else if (slaDiff > (THRESHOLD-1) && slaDiff < THRESHOLD)
```

File: packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx [hunk 6]
```diff
@@ -131,22 +120,21 @@ export const ProtectedPVCsSection: React.FC<ProtectedPVCsSectionProps> = ({
...
+  const [rpo, setRPO] = React.useState('N/A');
+  const clearSetIntervalId = React.useRef<NodeJS.Timeout>();
+  const updateRPO = React.useCallback(() => {
+    setRPO(
+      fromNow(selectedAppSet?.placementInfo?.[0]?.lastGroupSyncTime) || 'N/A'
     );
-    const rpo = Number(item?.value[1]);
-    return !!rpo ? humanizeRPO(rpo)?.string : 'N/A';
-  }, [selectedAppSet, lastSyncTimeData]);
+  }, [selectedAppSet]);
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx:131]:
```suggestion
  }, [lastGroupSyncTime]);
```

File: packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx [hunk 7]
```diff
@@ -131,22 +120,21 @@ export const ProtectedPVCsSection: React.FC<ProtectedPVCsSectionProps> = ({
...
-    const placementInfo = selectedAppSet?.placementInfo?.[0];
-    const item = lastSyncTimeData?.data?.result?.find(
-      (item: PrometheusResult) =>
-        item?.metric?.namespace === placementInfo?.drpcNamespace &&
-        item?.metric?.name === placementInfo?.drpcName
+  const [rpo, setRPO] = React.useState('N/A');
+  const clearSetIntervalId = React.useRef<NodeJS.Timeout>();
+  const updateRPO = React.useCallback(() => {
+    setRPO(
+      fromNow(selectedAppSet?.placementInfo?.[0]?.lastGroupSyncTime) || 'N/A'
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx:129]: is this info coming from DRPC ?

Comment (GowthamShanmugam) [reply packages/mco/components/mco-dashboard/data-policy/cluster-app-card/argo-application-set.tsx:129]: yes, DRPC has the last sync time, the same value is converted by the ramen hub operator into metrics

**Pull request merged**